lib/set-script.js

```
@@ -0,0 +1,48 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function(args) {
+    // Parse arguments
+    if (args.length !== 2) {
+        log.error('set-script', 'Expected 2 arguments: got ' + args.length)
+        return
+    }
+
+    // Set the script
+    let manifest
+    let warn = false
+    try {
+        manifest = fs.readFileSync(localPrefix + '/package.json', 'utf-8')
+    } catch {
+        log.error('set-script', 'package.json not found')
+        return
+    }
+    try {
+        manifest = JSON.parse(manifest)
+    } catch(error) {
+        log.error('set-script', 'Invalid package.json: ' + error)
+        return
```

same as before, `throw error` instead of returning 😊